api/src/main/java/org/geysermc/geyser/api/packs/GeyserResourcePackManifest.java

```diff
@@ -0,0 +1,241 @@
+         *
+         * @return the version string
+         */
+        String versionString();
+    }
+
+    /**
+     * Represents a module of a resource pack.
+     */
+    interface Module {
+
+        /**
+         * Gets the description of the module.
+         *
+         * @return the description
+         */
+        String description();
+
+        /**
+         * Gets the name of the module.
+         *
+         * @return the name
+         */
+        String name();
+
+        /**
+         * Gets the UUID of the module.
+         *
+         * @return the UUID
+         */
+        UUID uuid();
+
+        /**
+         * Gets the version of the module.
+         *
+         * @return the version
+         */
+        int[] version();
+    }
+
+    /**
+     * Represents a dependency of a resource pack.
+     */
+    interface Dependency {
+
+        /**
+         * Gets the description of the dependency.
+         *
+         * @return the description
+         */
+        UUID uuid();
+
+        /**
+         * Gets the name of the dependency.
+         *
+         * @return the name
+         */
+        int[] version();
+    }
+
+    /**
+     * Represents a resource pack version.
+     */
+    interface Version {
+
+        /**
+         * Gets the major version.
+         *
+         * @return the major version
+         */
+        int major();
+
+        /**
+         * Gets the minor version.
+         *
+         * @return the minor version
+         */
+        int minor();
+
+        /**
+         * Gets the patch version.
+         *
+         * @return the patch version
+         */
+        int patch();
+
+        /**
+         * Creates a new version from an array.
+         *
+         * @param ver the array
+         * @return the version
+         */
+        static Version fromArray(int[] ver) {
+            return new Version() {
+                @Override
+                public int major() {
+                    return ver[0];
+                }
+
+                @Override
+                public int minor() {
+                    return ver[1];
+                }
+
+                @Override
+                public int patch() {
+                    return ver[2];
+                }
+            };
+        }
+
+        static Version fromString(String ver) {
+            String[] split = ver.replace(']', ' ')
+                    .replace('[', ' ')
+                    .replaceAll(" ", "").split(",");
+
+            return new Version() {
+                @Override
+                public int major() {
+                    return Integer.parseInt(split[0]);
+                }
+
+                @Override
+                public int minor() {
+                    return Integer.parseInt(split[1]);
+                }
+
+                @Override
+                public int patch() {
+                    return Integer.parseInt(split[2]);
+                }
+            };
```

Comment:
I think this could be replaced with this, less repeated code.
```suggestion
            return fromArray(new int[]{Integer.parseInt(split[0]), Integer.parseInt(split[1]), Integer.parseInt(split[2]));   
```